src/main.rs

```diff
@@ -394,7 +408,61 @@ async fn run() -> Result<()> {
         PathBuf::from(&src)
     };
 
-    let mut choices = Vec::new();
+    // Let's first find all subdirectories that are compatible with our supported recipes.
+    let root_src_dir = src_dir.clone();
+    let mut open_subdirs = vec![(0, src_dir)];
+    let mut sourceroot_candidates = vec![];
+    while !open_subdirs.is_empty() {
+        // If we call this, we have a non-empty vector.
+        let (depth, target_dir) = open_subdirs.pop().unwrap();
+        let subdirs = std::fs::read_dir(&target_dir)
+            .context(format!("failed to list subdirectories of {}", target_dir.to_string_lossy()))?;
+
+        for maybe_subdir in subdirs {
+            let subdir = maybe_subdir.context(
+                format!("unexpected io error while reading a subdirectory of {}", target_dir.to_string_lossy())
+            )?;
+
+            // We want only subdirectories.
+            if !subdir.path().is_dir() {
+                continue;
+            }
+
+            if is_target_for_packaging(subdir.path()) {
+                sourceroot_candidates.push(subdir.path());
```

Comment:
we might be able to use `file_name` here instead, so we don't have to do `strip_prefix` later. the same applies to `open_subdirs`

Reply:
`file_name` only returns the final component, not the N - 1 final components